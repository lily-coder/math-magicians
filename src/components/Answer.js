/* eslint no-unused-vars: 0 no-undef: 0 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

const isNull = (prop) => {
  if (!prop) {
    return '0';
  }
  return prop;
};

const current = (total, next) => {
  if (total && next) {
    return isNull(next);
  }
  if (total) {
    return total;
  }
  return isNull(next);
};

const Answer = (props) => {
  const { total, next } = props;

  return (
      <div className='calc-row-one answer'>
        <p>{ current(total, next) }</p>
      </div>
  );
};

Answer.defaultProps = {
  total: null,
  next: null,
};

Answer.propTypes = {
  total: PropTypes.string,
  next: PropTypes.string,
};

export default Answer;
