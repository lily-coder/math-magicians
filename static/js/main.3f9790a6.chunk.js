(this["webpackJsonpmath-magicians"]=this["webpackJsonpmath-magicians"]||[]).push([[0],{14:function(t,e,n){},33:function(t,e,n){"use strict";n.r(e);var a=n(1),c=n.n(a),i=n(17),s=n.n(i),l=n(8),o=(n(14),n(4)),r=n(11),u=n(2),j=n(0),d=function(t){var e=function(e){var n=t.handleClick;n(e.target.value)};return Object(j.jsxs)("div",{className:"calculator-container",children:[Object(j.jsxs)("div",{className:"calc-row-two",children:[Object(j.jsx)("button",{value:"AC",className:"operator",onClick:e,children:"AC"}),Object(j.jsx)("button",{value:"+/-",className:"operator",onClick:e,children:"+/-"}),Object(j.jsx)("button",{value:"%",className:"operator",onClick:e,children:"%"}),Object(j.jsx)("button",{value:"\xf7",className:"divide operator",onClick:e,children:"\xf7"})]}),Object(j.jsx)("div",{className:"calc-row-three orange-right",children:Object(j.jsx)("button",{value:"x",className:"orange operator",onClick:e,children:"x"})}),Object(j.jsx)("div",{className:"calc-row-four orange-right",children:Object(j.jsx)("button",{value:"-",className:"orange operator",onClick:e,children:"-"})}),Object(j.jsx)("div",{className:"calc-row-five orange-right",children:Object(j.jsx)("button",{value:"+",className:"orange operator",onClick:e,children:"+"})})]})};d.defaultProps={handleClick:null};var m=d,b=function(t){return t||"0"},x=function(t,e){return t&&e?b(e):t||b(e)},h=function(t){var e=t.total,n=t.next;return Object(j.jsx)("div",{className:"calc-row-one answer",children:Object(j.jsx)("p",{children:x(e,n)})})};h.defaultProps={total:null,next:null};var O=h,v=function(t){var e=function(e){var n=t.handleClick;n(e.target.value)};return Object(j.jsxs)("div",{className:"calculator-container numbers-container",children:[Object(j.jsxs)("div",{className:"calc-row-five numbers-five",children:[Object(j.jsx)("button",{value:"1",onClick:e,children:"1"}),Object(j.jsx)("button",{value:"2",onClick:e,children:"2"}),Object(j.jsx)("button",{value:"3",onClick:e,children:"3"})]}),Object(j.jsxs)("div",{className:"calc-row-four numbers-four",children:[Object(j.jsx)("button",{value:"4",onClick:e,children:"4"}),Object(j.jsx)("button",{value:"5",onClick:e,children:"5"}),Object(j.jsx)("button",{value:"6",onClick:e,children:"6"})]}),Object(j.jsxs)("div",{className:"calc-row-three numbers-three",children:[Object(j.jsx)("button",{value:"7",onClick:e,children:"7"}),Object(j.jsx)("button",{value:"8",onClick:e,children:"8"}),Object(j.jsx)("button",{value:"9",onClick:e,children:"9"})]}),Object(j.jsxs)("div",{className:"calc-row-six numbers-six",children:[Object(j.jsx)("button",{className:"zero",value:"0",onClick:e,children:"0"}),Object(j.jsx)("button",{className:"dot",value:".",onClick:e,children:"."})]})]})};v.defaultProps={handleClick:null};var p=v,f=function(t){return Object(j.jsx)("div",{className:"calc-row-six equal-container",children:Object(j.jsx)("button",{value:"=",className:"equal orange",onClick:function(e){var n=t.handleClick;n(e.target.value)},children:"="})})};f.defaultProps={handleClick:null};var g=f,C=n(15),N=n.n(C);function k(t,e,n){var a=N()(t),c=N()(e);if("+"===n)return a.plus(c).toString();if("-"===n)return a.minus(c).toString();if("x"===n)return a.times(c).toString();if("\xf7"===n)return a.div(c).toString();if("%"===n)return a.mod(c).toString();throw Error("Unknown operation '".concat(n,"'"))}function q(t,e){return"AC"===e?{total:null,next:null,operation:null}:e.match(/[0-9]+/)?"0"===e&&"0"===t.next?{}:t.operation?t.next?Object(o.a)(Object(o.a)({},t),{},{next:t.next+e}):Object(o.a)(Object(o.a)({},t),{},{next:e}):t.next?{next:t.next+e,total:null}:{next:e,total:null}:"."===e?t.next?t.next.includes(".")?Object(o.a)({},t):Object(o.a)(Object(o.a)({},t),{},{next:"".concat(t.next,".")}):t.operation?{next:"0."}:t.total?t.total.includes(".")?{}:{total:"".concat(t.total,".")}:{total:"0."}:"="===e?t.next&&t.operation?{total:k(t.total,t.next,t.operation),next:null,operation:null}:{}:"+/-"===e?t.next?Object(o.a)(Object(o.a)({},t),{},{next:(-1*parseFloat(t.next)).toString()}):t.total?Object(o.a)(Object(o.a)({},t),{},{total:(-1*parseFloat(t.total)).toString()}):{}:t.next||!t.total||t.operation?t.next?t.operation?t.total&&!t.next?Object(o.a)(Object(o.a)({},t),{},{operation:e}):{total:k(t.total,t.next,t.operation),next:null,operation:e}:{total:t.next,next:null,operation:e}:{operation:e}:Object(o.a)(Object(o.a)({},t),{},{operation:e})}var w=function(){var t=Object(a.useState)(!1),e=Object(r.a)(t,2),n=(e[0],e[1]);return Object(j.jsxs)("nav",{className:"navBar",children:[Object(j.jsx)("h1",{className:"math-magicians-header",children:"Math Magicians"}),Object(j.jsx)("ul",{className:"navbar",children:[{id:1,path:"/home",text:"Home"},{id:2,path:"/calculator",text:"Calculator"},{id:3,path:"/quote",text:"Quote"}].map((function(t){return Object(j.jsx)("li",{children:Object(j.jsx)(l.b,{to:t.path,activeClassName:"active-link",onClick:function(){n(!1)},exact:!0,children:t.text})},t.id)}))})]})},S=function(){Object(u.f)();return Object(j.jsx)("div",{className:"quote-content",children:Object(j.jsx)("p",{children:"Mathematics is not about numbers, equations, computations, or algorithms: its about understanding- William Paul Thurston"})})},E=function(){Object(u.f)();return Object(j.jsxs)("div",{className:"home-content",children:[Object(j.jsx)("h2",{className:"home-h2",children:"Welcome to our page!"}),Object(j.jsx)("p",{children:"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut lobortis quis sem quis tincidunt. Donec ut risus a enim consequat aliquet. Curabitur et metus vitae nunc porttitor accumsan vitae at magna. Nunc scelerisque ante est, eget tincidunt ipsum sodales non. Nunc maximus, urna ut finibus lobortis, erat orci tristique massa, semper elementum lorem felis sit amet massa. Etiam elementum fringilla odio, ac dignissim orci pulvinar quis. Cras tempor, sem sed commodo elementum, odio elit venenatis nunc, nec pellentesque justo sapien ut metus. Curabitur quis tellus ac odio efficitur ultrices eget sit amet risus. Quisque commodo augue nisl, sed consectetur dui vestibulum sed."}),Object(j.jsx)("p",{children:"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut lobortis quis sem quis tincidunt. Donec ut risus a enim consequat aliquet. Curabitur et metus vitae nunc porttitor accumsan vitae at magna. Nunc scelerisque ante est, eget tincidunt ipsum sodales non. Nunc maximus, urna ut finibus lobortis, erat orci tristique massa, semper elementum lorem felis sit amet massa. Etiam elementum fringilla odio, ac dignissim orci pulvinar quis. Cras tempor, sem sed commodo elementum, odio elit venenatis nunc, nec pellentesque justo sapien ut metus. Curabitur quis tellus ac odio efficitur ultrices eget sit amet risus. Quisque commodo augue nisl, sed consectetur dui vestibulum sed."})]})},I=function(){var t={total:null,next:null,operation:null},e=Object(a.useState)(t),n=Object(r.a)(e,2),c=n[0],i=n[1],s=Object(a.useState)({status:!1}),l=Object(r.a)(s,2),d=l[0],b=l[1],x=function(t){return i((function(e){return Object(o.a)(Object(o.a)({},e),t)}))};Object(a.useEffect)((function(){if("savedInput"in d){b({status:!1});var t=q(c,d.savedInput);x(t)}}),[d]);var h=function(e){if(d.status)return b({savedInput:e}),void i(t);var n;try{n=q(c,e)}catch(a){b({status:!0}),i({total:"Invalid operation: Can't Divide by Zero",next:null,operation:null})}x(n)};return Object(j.jsxs)(j.Fragment,{children:[Object(j.jsx)(w,{}),Object(j.jsxs)(u.c,{children:[Object(j.jsx)(u.a,{exact:!0,path:"/calculator",children:Object(j.jsxs)("div",{className:"calc-contain",children:[Object(j.jsx)("h2",{className:"calc-header",children:"Let's do some math"}),Object(j.jsxs)("div",{className:"calculator-container",children:[Object(j.jsx)(m,{handleClick:h}),Object(j.jsx)(O,{total:c.total,next:c.next}),Object(j.jsx)(p,{handleClick:h}),Object(j.jsx)(g,{handleClick:h})]})]})}),Object(j.jsx)(u.a,{path:"/home",children:Object(j.jsx)(E,{})}),Object(j.jsx)(u.a,{path:"/quote",children:Object(j.jsx)(S,{})})]})]})};s.a.render(Object(j.jsx)(c.a.StrictMode,{children:Object(j.jsx)(l.a,{basename:"/math-magicians",children:Object(j.jsx)(I,{})})}),document.getElementById("root"))}},[[33,1,2]]]);
//# sourceMappingURL=main.3f9790a6.chunk.js.map