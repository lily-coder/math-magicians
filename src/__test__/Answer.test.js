/* eslint no-unused-vars: 0 no-undef: 0 */
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect.js';
import Answer from '../components/Answer.js';

describe('Answer', () => {
  test('Correctly renders Answer component', () => {
    const answerTree = renderer.create(<Answer/>).toJSON();
    expect(answerTree).toMatchSnapshot();
  });
});