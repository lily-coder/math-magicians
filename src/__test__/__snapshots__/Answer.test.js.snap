// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer Correctly renders Answer component 1`] = `
<div
  className="calc-row-one answer"
>
  <p>
    0
  </p>
</div>
`;
